### PR TITLE
fix: BYTES column reflection failure

### PR DESCRIPTION
**Pull request by IlyaFaer**

### CODE REVIEW

**Comment by IlyaFaer** [comment]
While working on benchmarks I've notice an error with BYTES type reflection. Adding a test reproducing the failure, fixing the error.

File: google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py [hunk 1]
```diff
@@ -441,6 +442,11 @@ def get_columns(self, connection, table_name, schema=None, **kw):
                     end = col[1].index(")")
                     size = int(col[1][7:end])
                     type_ = _type_map["STRING"](length=size)
+                # add test creating a table with bytes
+                elif col[1].startswith("BYTES"):
+                    end = col[1].index(")")
+                    size = int(col[1][6:end])
+                    type_ = _type_map["BYTES"](length=size)
```
Comment (IlyaFaer) [inline google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py:449]: Dealing with BYTES in the same way the `String` type is processed

**Review by larkee — APPROVED** [review]
LGTM. Were there no `BYTES`-specific tests in the test suite?

**Comment by IlyaFaer** [comment]
> LGTM. Were there no `BYTES`-specific tests in the test suite?

Yeah, I didn't find any, so added several myself.

**Pull request merged**